resources/js/multi-step-form.js

```diff
@@ -0,0 +1,166 @@
+            } else {
+                el.removeAttribute('data-active');
+                el.removeAttribute('data-done');
+            }
+        });
+    }
+    validateStep(step) {
+        this.form.querySelectorAll('.js-error').forEach(e => e.remove());
+        this.form.querySelectorAll('.border-red-500').forEach(e => e.classList.remove('border-red-500'));
+        switch(step) {
+            case 1:
+                return this.validateFields({
+                    fields: ['property_type'],
+                    stepSelector: '.step[data-step="1"]',
+                    type: 'radio',
+                    errorSelector: '.step-error',
+                    errorMessage: '物件種別を選択してください',
+                });
+            case 2:
+                return this.validateFields({
+                    fields: ['postcode', 'prefecture', 'address'],
+                    stepSelector: '.step[data-step="2"]',
+                    type: 'input',
+                    errorMessages: {
+                        postcode: '郵便番号を入力してください',
+                        prefecture: '都道府県を選択してください',
+                        address: '住所を入力してください'
+                    }
+                });
+            case 3:
+                return this.validateFields({
+                    fields: ['company'],
+                    stepSelector: '.step[data-step="3"]',
+                    type: 'input',
+                    errorMessages: {
+                        company: 'ガス会社名を入力してください'
+                    },
+                    errorClass: 'mt-2'
+                });
+            case 4:
+                return this.validateFields({
+                    fields: ['name', 'phone', 'email'],
+                    stepSelector: '.step[data-step="4"]',
+                    type: 'input',
+                    errorMessages: {
+                        name: 'お名前を入力してください',
+                        phone: '電話番号を入力してください',
+                        email: 'メールアドレスを入力してください'
+                    }
+                });
+            default:
+                return true;
+        }
+    }
+    validateFields({ fields, stepSelector, type, errorSelector, errorMessage, errorMessages, errorClass }) {
+        let valid = true;
+        if (type === 'radio') {
+            const selected = this.form.querySelector(`[name="${fields[0]}"]:checked`);
+            const stepError = this.form.querySelector(`${stepSelector} ${errorSelector}`);
+            if (!selected) {
+                this.propertyTypeInputs.forEach(radio => {
+                    radio.parentElement.classList.add('border-red-500');
+                });
+                if (stepError) {
+                    stepError.innerText = errorMessage;
+                    stepError.classList.remove('hidden');
+                }
+                valid = false;
+            } else {
+                if (!stepError?.classList.contains('hidden')) {
+                    stepError?.classList.add('hidden');
+                }
+            }
```

Comment:
@baovnt-zigexn 
Hạn chế lồng quá nhiều if => Khó đọc code.
```suggestion
            } else if (!stepError?.classList.contains('hidden')) {
                 stepError?.classList.add('hidden');
            }
```